File: Phase08/Phase08/TestPhase08/InvertedIndexTest.cs
using System.Collections.Generic;
using Phase08;
using Xunit;

namespace TestPhase08
{
    public class InvertedIndexTest
    {
        private readonly InvertedIndex _invertedIndex;
        private readonly Dictionary<string,string> _docNameMapToContent;


        public InvertedIndexTest()
        {
            _invertedIndex = new InvertedIndex();
            _docNameMapToContent = new Dictionary<string, string>
            {
                {"text1", "one two"}, {"text2", "five six seven eight nine"}, {"text3", "one two three "}
            };

        }

        [Fact]
        public void TestBuildInvertedIndex_WordExistInADoc() {
            _invertedIndex.BuildInvertedIndex(_docNameMapToContent);
            Assert.Equal(new HashSet<string>(){"text2"}, 
                _invertedIndex.GetInvertedIndexValue("six"));

        }

        [Fact]
        public void TestBuildInvertedIndex_WordExistInTwoDoc() {
            _invertedIndex.BuildInvertedIndex(_docNameMapToContent);
            Assert.Equal(new HashSet<string>(){"text3", "text1"},
                _invertedIndex.GetInvertedIndexValue("one"));

        }

        [Fact]
        public void TestBuildInvertedIndex_WordNotExistInDoc() {
            _invertedIndex.BuildInvertedIndex(_docNameMapToContent);
            Assert.Equal(new HashSet<string>(),
                _invertedIndex.GetInvertedIndexValue("ten"));

        }
    }
}

File: Phase03/Tests/InvertedIndexTest.java
package Tests;

import java.util.HashMap;
import java.util.Map;

import Phase02.InvertedIndex;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;


public class InvertedIndexTest {

    private InvertedIndex invertedIndex;
    private Map<String, String> docNameMapToContent;

    @Before
    public void setup() {
        invertedIndex = new InvertedIndex();
        docNameMapToContent = new HashMap<>();

        docNameMapToContent.put("text1", "one two");
        docNameMapToContent.put("text2", "five six seven eight nine");
        docNameMapToContent.put("text3", "one two three ");
    }

    @Test
    public void testInvertedIndexWordExistInADoc() {
        invertedIndex.buildInvertedIndex(docNameMapToContent);
        Assert.assertArrayEquals(new String[]{"text2"},
                invertedIndex.getInvertedIndexValue("six").toArray());

    }

    @Test
    public void testInvertedIndexWordExistInTwoDoc() {
        invertedIndex.buildInvertedIndex(docNameMapToContent);
        Assert.assertArrayEquals(new String[]{"text3", "text1"},
                invertedIndex.getInvertedIndexValue("one").toArray());

    }

    @Test
    public void testInvertedIndexWordNotExistInDoc() {
        invertedIndex.buildInvertedIndex(docNameMapToContent);
        Assert.assertArrayEquals(new String[]{},
                invertedIndex.getInvertedIndexValue("ten").toArray());

    }

}
